rocky/keiko.py

```diff
@@ -24,17 +24,16 @@ def generate_report(self, template: str, data: Dict, glossary: str) -> str:
         res.raise_for_status()
         return res.json()["report_id"]
 
-    def get_report(self, report_id: str) -> Optional[bytes]:
+    def get_report(self, report_id: str) -> bytes:
 
         # try max 15 times to get the report, 1 second interval
         for i in range(15):
+            time.sleep(1)
             res = self.session.get(f"{self._base_uri}/reports/{report_id}.keiko.pdf")
             if res.status_code == 200:
                 return res.content
-            else:
-                time.sleep(1)
-
-        return None
+            if i == 14:
```

Comment:
This `if` is redundant, you can make the `res.raise_for_status()` call at the end of the body after the loop finishes

Reply:
Yeah, it's ugly, sorry. Ran into this warning and wasn't sure about the cleanest solution...

![Screenshot 2023-01-06 at 10 00 52](https://user-images.githubusercontent.com/9639395/210967430-59f2c074-1491-4eb7-82bd-43a43ff3a8e3.png)

Reply:
In that case you could use the `else` clause:
```python3
for ...:
    pass
else:
    pass  # executed when the loop is completed without any break
```

Reply:
Fixed.